src/vs/workbench/contrib/debug/browser/debugSession.ts

```diff
@@ -108,6 +110,11 @@ export class DebugSession implements IDebugSession {
 		return includeRoot && this.root ? `${this.configuration.name} (${resources.basenameOrAuthority(this.root.uri)})` : this.configuration.name;
 	}
 
+	setName(name: string): void {
+		this.configuration.name = name;
```

Comment:
I do not think we should change the underlying `configuration` object.
This to me feels like we should introduce a new attribute on the session called `name`.
And in the `getLabel` we use the `name` of the session, if the `name` is not defined then go back to providing the `configuration.name` as a fallback.

Reply:
That way we have two independt attributes, and we do not mix concepts between the name of the configuration and the name of the session.
Let me know what you think.

Reply:
Yes, that's the approach we should use.

Reply:
Sounds good, updated.